confignode/src/main/java/org/apache/iotdb/confignode/manager/load/LoadManager.java

```diff
@@ -128,34 +145,66 @@ public Map<String, DataPartitionTable> allocateDataPartition(
     return partitionBalancer.allocateDataPartition(unassignedDataPartitionSlotsMap);
   }
 
-  @Override
-  public void run() {
-    int balanceCount = 0;
-    while (true) {
-      try {
-
-        if (getConsensusManager().isLeader()) {
-          // Send heartbeat requests to all the online DataNodes
-          pingOnlineDataNodes(getNodeManager().getOnlineDataNodes(-1));
-          // TODO: Send heartbeat requests to all the online ConfigNodes
-
-          // Do load balancing
-          doLoadBalancing(balanceCount);
-          balanceCount += 1;
-        } else {
-          // Discard all cache when current ConfigNode is not longer the leader
-          heartbeatCacheMap.clear();
-          balanceCount = 0;
-        }
-
-        TimeUnit.MILLISECONDS.sleep(heartbeatInterval);
-      } catch (InterruptedException e) {
-        LOGGER.error("Heartbeat thread has been interrupted, stopping ConfigNode...", e);
-        System.exit(-1);
+  /** Start the heartbeat service */
+  public void start() {
+    LOGGER.debug("Start Heartbeat Service of LoadManager");
+    synchronized (heartbeatMonitor) {
+      if (currentHeartbeatFuture == null) {
+        currentHeartbeatFuture =
+            ScheduledExecutorUtil.unsafelyScheduleWithFixedDelay(
+                heartBeatExecutor,
+                this::heartbeatLoopBody,
+                0,
+                heartbeatInterval,
+                TimeUnit.MILLISECONDS);
       }
     }
   }
 
+  /** Stop the heartbeat service */
+  public void stop() {
+    LOGGER.debug("Stop Heartbeat Service of LoadManager");
+    synchronized (heartbeatMonitor) {
+      if (currentHeartbeatFuture != null) {
+        currentHeartbeatFuture.cancel(false);
+        currentHeartbeatFuture = null;
+      }
+    }
+  }
+
+  /**
+   * Running state of heartbeat service
+   *
+   * @return true if heartbeat service is running, false otherwise
+   */
+  public boolean isHeartbeatRunning() {
+    return currentHeartbeatFuture != null;
+  }
+
+  /** loop body of the heartbeat thread */
+  private void heartbeatLoopBody() {
+    if (getConsensusManager().isLeader()) {
+      // Send heartbeat requests to all the online DataNodes
+      pingOnlineDataNodes(getNodeManager().getOnlineDataNodes(-1));
+      // TODO: Send heartbeat requests to all the online ConfigNodes
+
+      // Do load balancing
+      doLoadBalancing(balanceCount);
+      balanceCount += 1;
+    }
+  }
+
+  /**
+   * Handler when the heartbeat thread is down
+   *
+   * @param t thread
+   * @param e exception
+   */
+  private void loadExceptionHandler(Thread t, Throwable e) {
+    LOGGER.error("Heartbeat thread has been interrupted, stopping ConfigNode...", e);
+    System.exit(-1);
```

Comment:
Just logging an error is enough. Do not kill ConfigNode by ourselves.

Reply:
remove the exception handler since `safelyScheduleWithFixedDelay` has already print the error log